gem_metrics/__init__.py

```diff
@@ -112,84 +198,94 @@ def process_submission(outs: Submission, refs: Optional[Dict], srcs: Optional[Di
 
     Returns a dict keyed by dataset names, containing the dicts for each dataset's results.
     """
-    values = {'submission_name': outs.name,
-              'param_count': outs.param_count}
+    # Handle the CPU-bound metrics in parallel to speed up computation.
+    manager = Manager()
+    shared_dict = manager.dict()
+    shared_dict["submission_name"] = outs.name
+    shared_dict["param_count"] = outs.param_count
+
+    def multiprocess_compute(dataset, outs_ds, refs_ds, srcs_ds, metrics_dict, cache):
+        shared_dict[dataset] = compute(
+            outs_ds, refs_ds, srcs_ds, metrics_dict, cache, dataset
+        )
+
+    job_args = []
+
     for dataset in outs.datasets:
-        logger.info(f'Computing metrics for {dataset}...')
+        logger.info(f"Computing metrics for {dataset}...")
         outs_ds = outs.predictions_for(dataset)
-        # use default reference files if no custom ones are provided
-        refs_ds = refs[dataset] if (refs and dataset in refs) else load_references(dataset)
-        srcs_ds = srcs[dataset] if (srcs and dataset in srcs) else load_sources(dataset)
-        values[dataset] = compute(outs_ds, refs_ds, srcs_ds, metrics_dict=metrics_dict)
-    return values
+        refs_ds = refs.get(dataset, None)
+        srcs_ds = srcs.get(dataset, None)
+        job_args.append(
+            (dataset, outs_ds, refs_ds, srcs_ds, parallel_metric_dict, cache)
+        )
+
+    pool = Pool(processes=12)
```

Comment:
What would this do with much fewer cores? How hard would it be to parameterize it?

Reply:
I can add a parameter for this. In theory it should work with <12 cores as well though.